// material-ui-responsive-menu/es/ResponsiveMenu.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _this = this;

import React from 'react';
import PropTypes from 'prop-types';
import ContainerDimensions from 'react-container-dimensions';
import IconButton from 'material-ui/IconButton/IconButton';
import ListMenu from './ListMenu.js';

var styles = {
  menu_container: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    margin: 0,
    padding: 0
  }
};

var getIcon = function getIcon(item, i) {
  if (item.seperator) {
    return undefined;
  } else if (item.menu) {
    return item.getSelectMenuItems();
  } else {
    return React.createElement(
      IconButton,
      {
        key: i,
        disabled: item.disabled,
        onClick: item.onClick,
        tooltip: item.tooltip },
      item.icon
    );
  }
};

var Menu = function Menu(props) {
  var iconMenuColor = props.iconMenuColor,
      width = props.width,
      marginCorrection = props.marginCorrection,
      iconWidth = props.iconWidth,
      menuList = props.menuList;


  var icons = [];
  var menus = [];

  var visibleItems = menuList.filter(function (item) {
    return item.hidden !== true;
  });
  var iconW = iconWidth !== undefined ? iconWidth : 48;
  var correction = marginCorrection !== undefined ? marginCorrection : 0;
  var breackLimit = visibleItems.length * iconW;

  var hiddenWidth = breackLimit - width - correction;
  var hiddenIcons = Math.ceil(hiddenWidth / iconW) + 1;

  if (hiddenIcons > 0) {
    menus = visibleItems.slice(visibleItems.length - hiddenIcons, visibleItems.length), icons = visibleItems.slice(0, hiddenIcons * -1);
  } else {
    menus = [];
    icons = visibleItems;
  }

  return React.createElement(
    'div',
    {
      style: styles.menu_container,
      ref: function ref(container) {
        _this.menuContainer = container;
      } },
    icons.map(function (item, i) {
      return getIcon(item, i);
    }),
    menus.length != 0 && React.createElement(ListMenu, {
      items: menus,
      iconMenuColor: iconMenuColor
    })
  );
};

var ResponsiveMenu = function ResponsiveMenu(props) {
  return React.createElement(
    ContainerDimensions,
    null,
    function (_ref) {
      var width = _ref.width;
      return React.createElement(Menu, _extends({ width: width }, props));
    }
  );
};

ResponsiveMenu.propTypes = process.env.NODE_ENV !== "production" ? {
  menuList: PropTypes.array.isRequired,
  iconWidth: PropTypes.number,
  marginCorrection: PropTypes.number,
  iconMenuColor: PropTypes.string
} : {};

export default ResponsiveMenu;
